packages/cli/src/cmds/compare/ChangeReporter.ts

```diff
@@ -155,30 +148,73 @@ export default class ChangeReporter {
       console.log(
         mutedStyle(`AppMap ${revisionName}/${appmap} is unreferenced so it will be deleted.`)
       );
-      const path = this.appmapData.appmapPath(revisionName, appmap);
+      const path = this.paths.appmapPath(revisionName, appmap);
       const fileName = [path, 'appmap.json'].join('.');
       assert(await exists(fileName));
       await rm(fileName);
       await rm(path, { recursive: true });
     };
 
     for (const revisionName of [RevisionName.Base, RevisionName.Head]) {
-      (await this.appmapData.appmaps(revisionName)).forEach((appmap) => {
+      (await this.paths.appmaps(revisionName)).forEach((appmap) => {
         if (!this.referencedAppMaps.test(revisionName, appmap)) deleteAppMap(revisionName, appmap);
       });
     }
   }
 
-  async report(): Promise<ChangeReport> {
-    const generator = new ReportGenerator(this);
-    const apiDiff = await generator.apiDiff();
+  async report(options: ChangeReportOptions): Promise<ChangeReport> {
+    const { appMapMetadata, baseAppMaps, headAppMaps, failedAppMaps } = this;
+    assert(appMapMetadata);
+    assert(baseAppMaps);
+    assert(headAppMaps);
+    assert(failedAppMaps);
+
+    const generator = new ReportFieldCalculator(this);
+    const apiDiff = await generator.apiDiff(options.reportRemoved);
+
+    const isNewFn = isNew(baseAppMaps, isTest(appMapMetadata));
+    const isChangedFn = isChanged(baseAppMaps, isTest(appMapMetadata), this.digests);
```

Comment:
This is the meat of the fix here - `isNew` and  `isChanged` are only true if `isTest` is also true.